cmd/local-locker.go

```diff
@@ -227,11 +227,19 @@ func (l *localLocker) ForceUnlock(ctx context.Context, args dsync.LockArgs) (rep
 	default:
 		l.mutex.Lock()
 		defer l.mutex.Unlock()
-		if len(args.UID) != 0 {
-			return false, fmt.Errorf("ForceUnlock called with non-empty UID: %s", args.UID)
+		if len(args.UID) == 0 {
+			for _, resource := range args.Resources {
+				delete(l.lockMap, resource) // Remove the lock (irrespective of write or read lock)
+			}
+			return true, nil
 		}
-		for _, resource := range args.Resources {
-			delete(l.lockMap, resource) // Remove the lock (irrespective of write or read lock)
+
+		for _, lris := range l.lockMap {
+			for _, lri := range lris {
+				if lri.UID == args.UID {
+					l.removeEntry(lri.Name, dsync.LockArgs{Owner: lri.Owner, UID: lri.UID}, &lris)
+				}
+			}
```

Comment:
Not super happy about this linear search. Is this for manually unlocking from cli only?

Reply:
It was manual-only @klauspost but it looks like now that it can get used with Refresh() logic we might have to fix this.

Reply:
@vadmeste ^^

Reply:
> It was manual-only @klauspost but it looks like now that it can get used with Refresh() logic 
yes correct

> we might have to fix this.
@harshavardhana  Fix what exactly ?

Reply:
Mainly @klauspost is asking about the linear map search

Reply:
> Mainly @klauspost is asking about the linear map search

yes, that means we need to refactor all the code. This is not a hot path, this is only cleanup of not refreshed locks which should be usually be removed using Unlock/RUnlock